backend/DAO/messageDAO.js

```diff
@@ -0,0 +1,52 @@
+const database = require('../DB/connection');
+const SuperError = require('../errors').SuperError;
+const ERR_INTERNAL_SERVER_ERROR = require('../errors').ERR_INTERNAL_SERVER_ERROR;
+
+module.exports = {
+
+    saveNewMessage: async (messageBody) => {
+        let result = await database('message')
+            .returning()
+            .insert({
+                student_id: messageBody.studentId,
+                employer_id: messageBody.employerId,
+                message: messageBody.message,
+                send_date: messageBody.sendDate,
+                sender: messageBody.sender
+            },['student_id', 'employer_id', 'message', 'send_date', 'sender'])
+            .catch(error => {
+                if(error){
+                    throw new SuperError(ERR_INTERNAL_SERVER_ERROR, 'There has been a problem saving your message. Please try again');
+                }
+            });
+        return result[0];
+    },
+
+    getConversation: (studentId, employerId) => {
+        return database('message')
+            .select('student_id', 'employer_id', 'message', 'send_date', 'sender')
+            .where('student_id', studentId)
+            .andWhere('employer_id', employerId)
+            .orderBy('send_date', 'asc')
+            .catch(error => {
+                throw new SuperError(ERR_INTERNAL_SERVER_ERROR, 'There has been a problem looking for this conversation. Please try again');  
```

Comment:
@AnnaBergamasco

Reply:
actually the error is not thrown when the conversation does not exists (in that case there is no error and we just have a 200 OK with an empty body, and this make sense when we have a new match and there are still no messages between the two) but when we get some errors from knex. This can happen, for example, if we mess up something with migrations in the db (it happened to me once when I switched from PLAD-162 to PLAD-79).

Reply:
IDK if this answer your question

Reply:
@AnnaBergamasco I am not really sure, but the tests are failing, with this log: 
![image](https://user-images.githubusercontent.com/37336669/101763204-c2c87580-3ade-11eb-90fe-1e529acf1b8a.png)

Reply:
It is a mistake in the test that happens when there are no messages in the db. I will fix this.